source/Core/Drivers/Font.h

```diff
@@ -8,863 +8,2274 @@
-    0x8c, 0x94, 0xae, 0x80, 0xbe, 0x8e, 0xa6, 0x8e, 0xa6, 0x8e, 0xa6, 0x8e, 0xa6, 0x8a, 0xa6, 0x8a, 0xa6, 0x8a, 0xa6, 0x8a, 0x46, 0x4a, 0x22, 0x18, 0x07, 0x00, 0x00, 0x00};
-
-const uint8_t disconnectedTip[] = {
-    // width = 42
-    // height = 16
-    0x00, 0x00, 0x00, 0x80, 0x80, 0x80, 0xc0, 0x00, 0xc0, 0xc0, 0xc0, 0xc0, 0xc0, 0xc0, 0xc0, 0xcc, 0x9c, 0x38, 0x70, 0xe0, 0xc0, 0x80, 0x20, 0x70, 0x38, 0x1c, 0xcc, 0x40,
-    0x80, 0x00, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0xc0, 0x60, 0xe0, 0x00, 0x01, 0x01, 0x01, 0x02, 0x01, 0x02, 0x05, 0x00, 0x07, 0x01, 0x04, 0x01, 0x04, 0x01,
-    0x04, 0x31, 0x38, 0x1c, 0x0e, 0x04, 0x01, 0x03, 0x07, 0x0e, 0x1c, 0x39, 0x30, 0x01, 0x03, 0x00, 0x02, 0x01, 0x02, 0x01, 0x02, 0x01, 0x02, 0x01, 0x04, 0x09, 0x0f, 0x00};
-#endif
-
-const uint8_t buttonB[] = {
-    // width = 42
-    // height = 16
-    0x00, 0x00, 0x00, 0xe0, 0x18, 0x04, 0x02, 0x02, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x71, 0x55, 0x71, 0x01, 0x01, 0xfd, 0x01, 0x01, 0x81, 0xbd, 0x81, 0x01, 0x01,
-    0x01, 0x01, 0x01, 0x01, 0x01, 0x02, 0x02, 0x04, 0x18, 0xe0, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x07, 0x18, 0x20, 0x40, 0x40, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80,
-    0x80, 0x80, 0xbf, 0x80, 0x80, 0x8e, 0xaa, 0x8e, 0x80, 0x83, 0xba, 0x83, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x40, 0x40, 0x20, 0x18, 0x07, 0x00, 0x00, 0x00, 0x00};
-
-// const uint8_t brightnessIcon[] = {
-//     // width = 16
-//     // height = 16
-//     0x80, 0x86, 0x8E, 0x9C, 0x18, 0xC0, 0xE0, 0xEF, 0xEF, 0xE0, 0xC0, 0x18, 0x9C, 0x8E, 0x86, 0x80, 0x01, 0x61, 0x71, 0x39, 0x18, 0x03, 0x07, 0xF7, 0xF7, 0x07, 0x03, 0x18, 0x39, 0x71, 0x61, 0x01};
-
-// const uint8_t invertDisplayIcon[] = {
-//     // width = 24
-//     // height = 16
-//     0xFE, 0x01, 0x79, 0x25, 0x79, 0x01, 0xFE, 0x00, 0x20, 0x20, 0x20, 0x20, 0xDF, 0x07, 0x8F, 0xDF, 0xFF, 0x01, 0xFE, 0x86, 0xDA, 0x86, 0xFE, 0x01,
-//     0x7F, 0x80, 0xA4, 0xBE, 0xA0, 0x80, 0x7F, 0x00, 0x04, 0x0E, 0x1F, 0x04, 0xFB, 0xFB, 0xFB, 0xFB, 0xFF, 0x80, 0x7F, 0x5B, 0x41, 0x5F, 0x7F, 0x80};
-
-const uint8_t RepeatOnce[] = {
-    // width = 16
-    // height = 16
-    0x00, 0xc0, 0xf0, 0x78, 0x1c, 0x0c, 0x0e, 0x06, 0x06, 0x0e, 0x2c, 0x3c, 0x38, 0x3c, 0x00, 0x00,
-    0x00, 0x01, 0x08, 0x04, 0x7e, 0x00, 0x00, 0x60, 0x60, 0x70, 0x30, 0x38, 0x1e, 0x0f, 0x03, 0x00};
-
-const uint8_t RepeatInf[] = {
-    // width = 16
-    // height = 16
-    0x00, 0xc0, 0xf0, 0x78, 0x1c, 0x0c, 0x0e, 0x06, 0x06, 0x0e, 0x2c, 0x3c, 0x38, 0x3c, 0x00, 0x00,
-    0x00, 0x31, 0x49, 0x48, 0x30, 0x48, 0x48, 0x30, 0x00, 0x00, 0x30, 0x38, 0x1e, 0x0f, 0x03, 0x00};
-
-/*
- * 16x16 icons
- * 32 * 3 = Frame size * Frame count
- * */
-const uint8_t SettingsMenuIcons[][32 * 3] = {
-    // Power
-    // 3 frames
-    {
-        // Power 1st frame
-        // width = 16
+     * 21x32 icons
+     * 53 * 3 = Frame size * Frame count
+     * */
+    const uint8_t SettingsMenuIcons[][53 * 3] = {

```

Comment:
TL; DR: replace `53 * 3` by `252`. It will fix builds but I'm not sure if it will work on the hardware.

Longer version: I'm not an expert but a bit of write-up on that, if I may, to help. So, [log says](https://github.com/Ralim/IronOS/actions/runs/7159972697/job/19493827042?pr=1819) there is a compilation error. [Looking a bit above](https://github.com/Ralim/IronOS/actions/runs/7159972697/job/19493827042?pr=1819#step:8:41) there is a piece of additional information:
```
./Core/Drivers/Font.h:1523:5: error: too many initializers for 'const uint8_t [159]' {aka 'const unsigned char [159]'}
```
Basically, highlighted line above:
```
const uint8_t SettingsMenuIcons[][53 * 3]
```
tells that the second dimension of two-dimensional array `SettingsMenuIcons` is going to be stuffed with _at least_ `159` elements. However, the compilation error from log says that _actual_ amount of elements which are going to be stuffed into - _too many_.
Looking through [this version](https://github.com/discip/IronOS/blob/graphics-for-S60/P-%26-TS101/source/Core/Drivers/Font.h) of `Font.h` with the help of bash one liners with sed/grep/awk/etc, amount of elements (max) _roughly_ is `252`. Hence, replacing `53 * 3` by `252` _should_ do the trick. At least I did manage to reproduce the issue locally and then increasing the space for the second array up to `252` did make build successful again. However, I'm not sure about this _woodoo magic numbers_. But it seems that since this part is under `define` for specific models (with bigger OLEDs), I guess the other models shouldn't be affected anyway. :)

And, to be honest, now I'm really curious myself will firmware with this modification work on the real S60/TS101 or there will be _some_ glitches with display.

If some of that somehow correct, then the question now is how to properly arrange that transition from `53 * 3` to `252` from the point of code style.

It seems that this file is [at least partially] generated with some other scripts. If so, unfortunately this is well-known tradeoff for automagicalization: with static code you have to test & debug static code, with automagic code you have to test & debug scripts *AND* the generated code. ;)

I hope some of that will help anyway.

Reply:
Thank you for finding the culprit!
However, I'm not sure if it will work either. 😅